### DISCUSSION TITLE
如何更好的增加快捷键功能

### DISCUSSION THREAD

**Discussion by nihaojob** [ikuaitu/vue-fabric-editor#10] [Ideas]
参考了其他编辑器快捷键的功能，很多都是如下图的实现，会有一个中间层的Core，比如居中工具，然后把功能提供给快捷键与UI组件调用，这样的好处是可以把功能代码与UI、框架解耦，各个模块也会更独立。
![](https://cdn.nlark.com/yuque/0/2022/jpeg/443638/1669704066406-0f32cde8-8adb-441a-a8eb-b05553bedacd.jpeg)

目前的现状是下图所示，功能代码都包含在UI组件中，我们下一步的计划就是建立Core模块，先将与快捷键相关的功能逐步改造，把功能代码迁移到Core中，并管理上快捷键模块与UI组件。
![](https://cdn.nlark.com/yuque/0/2022/png/443638/1669704070393-2ab43402-3872-49e8-a5ee-3449244676a8.png)

**Comment by Qiu-Jun** [comment]
我重新优化了一下，不用每次新加小模块冗余代码

**Comment by nihaojob** [comment]
darkroomjs的插件化设计：https://github.com/MattKetmo/darkroomjs/blob/cd086fea7716cc6b6219a2df1f1501f63fbf30e5/lib/js/core/darkroom.js#L335
react-design-editor的handlers模块：
https://github.com/salgum1114/react-design-editor/blob/master/src/canvas/handlers/index.ts

**Reply by nihaojob** [reply]
重新优化快捷键模块代码
https://github.com/nihaojob/vue-fabric-editor/blob/main/src/core/initHotKeys.js
